stacks/api/function/webhook/src/webhook.controller.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  Post,
+  Put,
+  Query,
+  UseGuards
+} from "@nestjs/common";
+import {DEFAULT, NUMBER} from "@spica-server/core";
+import {Schema} from "@spica-server/core/schema";
+import {ObjectId, OBJECT_ID, DatabaseService} from "@spica-server/database";
+import {ActionGuard, AuthGuard} from "@spica-server/passport";
+import {Webhook} from "./interface";
+import {WebhookService} from "./webhook.service";
+
+@Controller("webhook")
+export class WebhookController {
+  constructor(private webhookService: WebhookService, private database: DatabaseService) {}
+
+  @Get("collections")
+  @UseGuards(AuthGuard(), ActionGuard("webhook:index"))
+  collections() {
+    return this.database
+      .collections()
+      .then(collections => collections.map(coll => coll.collectionName));
+  }
+
+  @Get()
+  @UseGuards(AuthGuard(), ActionGuard("webhook:index"))
+  find(
+    @Query("limit", DEFAULT(10), NUMBER) limit: number,
+    @Query("skip", DEFAULT(0), NUMBER) skip: number
+  ) {
+    const aggregate = [
+      {
+        $facet: {
+          meta: [{$count: "total"}],
+          data: [{$skip: skip}, {$limit: limit}]
+        }
+      },
+      {
+        $set: {
+          meta: {$ifNull: [{$arrayElemAt: ["$meta", 0]}, {total: 0}]}
+        }
+      }
+    ];
+    return this.webhookService.aggregate(aggregate).next();
```

Comment:
What's the reason of this `.next()` instead of `.toArray()` ?

Reply:
Cause we don't expect array of documents.